### PR TITLE
add CLA to contributing file  #148

### PR DESCRIPTION
**Pull request by nhsavage**
Update to add a CLA. Want sign off from all people who have contributed, and from Carwyn to check it meets requirements.

### CODE REVIEW

**Review by gredmond-mo — APPROVED** [review]
Look good to me

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,6 +2,43 @@
 
 You'd like to help? Great!  :tada:
 
+All contributions to `PyPRECIS` are made via merges with the develop branch.
+New contributors should add their details to the "code contributors" section of this document as part of their first request.
+The developer who reviews each pull request is responsible for checking that the contributor's name is listed in this file before merging the changes into master branch.
+
+## Code Contributors  
+
+ *  Hamish Steptoe (Met Office, UK), [hamish.steptoe@metoffice.gov.uk](mailto:hamish.steptoe@metoffice.gov.uk)
```
Comment (hsteptoe) [inline CONTRIBUTING.md:11]: I'd suggest we can link to github profiles (like with Iris [e.g.](https://scitools-iris.readthedocs.io/en/stable/whatsnew/index.html#iris-whatsnew)) and then remove emails so they can't be scraped for spam. Is there a particular reason we should share individual developer emails?  There is a generic PRECIS mailbox below and all other interaction should probably be via issues, pull requests or the [discussion board](https://github.com/MetOffice/PyPRECIS/discussions)
```suggestion
 *  @hsteptoe (Met Office, UK) 
```

**Comment by hsteptoe** [comment]
Is precis@metoffice.gov.uk still the best generic way to contact us about this repo?  Does the mailbox still exist? Do we need a specific PyPRECIS one?  Or do we need to update this guidance to make it clear the other ways you can contact us... maybe drive users towards the Discussion board...?

**Comment by rosannaamato** [comment]
Fine by me. The PRECIS mailbox still works but there's ongoing discussion about who monitors it.

File: CONTRIBUTING.md (hunk 1 above)
Comment (nhsavage) [reply CONTRIBUTING.md:11]: @cpelly  can you comment on this? I based this one off the example you linked

**Comment by nhsavage** [comment]
> Is [precis@metoffice.gov.uk](mailto:precis@metoffice.gov.uk) still the best generic way to contact us about this repo? Does the mailbox still exist? Do we need a specific PyPRECIS one? Or do we need to update this guidance to make it clear the other ways you can contact us... maybe drive users towards the Discussion board...?

This is a really good question, so I have opened a separate discussion for it so we don't block this PR.  #152

**Pull request merged**